### PR TITLE
Replace legacy admin bar menu with refactored version in production.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue #2241

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by benbowler** [comment]
Ready for code review, the Visual Regression tests to be fixed in #2594.

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good, but just needs a merge conflict resolved I think. Then it should be set 🙂

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks like an E2E test is failing—the `specs/modules/search-console/admin-bar.test.js` is frequently failing after me re-running it several times, so given the nature of this PR I assume it's related 😅 

Weird that some pass though. Are they all passing locally for you?

**Comment by benbowler** [comment]
Back with you @tofumatt. The E2E test issues were due to the general instability we have been seeing recently.

File: stories/adminbar.stories.js [hunk 1]
```diff
@@ -21,57 +21,905 @@
...
+					],
+					position: 2.564102564102564,
+				},
+			];
+			// "options": {
+			// 	"startDate": "2020-12-03",
+			// 	"endDate": "2021-01-27",
+			// 	"dimensions": "date",
+			// 	"url": "https://www.elasticpress.io/features/"
+			// }
```
Comment (tofumatt) [inline stories/adminbar.stories.js:565]:
```suggestion

```

File: stories/adminbar.stories.js [hunk 2]
```diff
@@ -21,57 +21,905 @@
...
+			// Set up the search console and analytics modules stores but provide no data.
+			provideModules( registry, [
+				{ slug: 'search-console', active: true, connected: true },
+				{ slug: 'analytics', active: true, connected: true },
+			] );
+
+			registry.dispatch( CORE_USER ).setReferenceDate( '2021-01-28' );
+
+			// Search Console mock data
+			const searchConsoleData = [
```
Comment (tofumatt) [inline stories/adminbar.stories.js:54]: This is a _lot_ of data to have in the story itself, let's move this outside the file to a fixture file or something to improve readability of this file.

File: stories/adminbar.stories.js [hunk 3]
```diff
@@ -21,57 +21,905 @@
...
-			'googlesitekit.Analytics',
-			addAnalyticsAdminbarWidget, 11 );
+			registry.dispatch( MODULES_SEARCH_CONSOLE ).receiveGetReport( searchConsoleData, { options: searchConsoleOptions } );
 
-		addFilter( 'googlesitekit.AdminbarModules',
-			'googlesitekit.SearchConsole',
-			addGoogleSitekitSearchConsoleAdminbarWidget );
+			// Mock Analytics mock data separately for each section
+			// Total Users
+			const analyticsTotalUsersData = [
```
Comment (tofumatt) [inline stories/adminbar.stories.js:577]: There's a lot of non-dynamic data in this file, can we move them out into fixtures/a separate file? It'd help if we ever wanna reuse it again, but mostly this data makes this file quite long and makes it hard to read the actual story logic.

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good, but let's move the fixture data outside the story itself to improve readability.

File: stories/adminbar.stories.js (hunk 2 above)
Comment (benbowler) [reply stories/adminbar.stories.js:54]: Moved to fixtures.

File: stories/adminbar.stories.js (hunk 3 above)
Comment (benbowler) [reply stories/adminbar.stories.js:571]: Moved to fixtures.

**Comment by benbowler** [comment]
Thanks Matt, I've moved those data into fixtures and removed the commented out code.

**Review by tofumatt — APPROVED** [review]
Awesome, thanks for moving those out. Looks good!

**Review by felixarntz — CHANGES_REQUESTED** [review]
Awesome work!

The PR misses only one thing, which is to delete all the old files that are now unused. I'll quickly do that.

**Review by felixarntz — APPROVED** [review]

**Pull request merged**